tee-worker/litentry/core/stf-task/receiver/src/lib.rs

```diff
@@ -136,6 +137,10 @@ where
 			return Ok(())
 		}
 
+		// swap the hash in the rpc connection registry to make sure furthre RPC responses go to
+		// the right channel
+		self.author_api.swap_rpc_connection_hash(*old_top_hash, top.hash());
```

Comment:
Yes I have TODOs in the code that maybe it's better to rely on `req_ext_hash` as the connection registry key - but it would require you to pass this hash all the way down to deep in top pool and change the listener/watch key as well.

I'm not yet sure if it would cause any problem because they are indeed two different tops (with different signers, different nonces), so they **should differ** from tx pool's perspective.

However, from the requester, we want it to behave as if it's a continuous, integral single request. By swapping the hash we basically say, hey I know it's technically two different trusted calls, but I want you two share the same channel to pass back responses - actually not bad I'd say 🤔 

What can be improved here is we'd better identify the other trusted calls by `req_ext_hash`, not by the value `0x01`

Reply:
@kziemianek  The `pre_xxx` and `post_xxx`, we had it before for indirect invocation. Do you mean to switch back?